src/sentry/snuba/metrics/extraction.py

```diff
@@ -496,11 +498,18 @@ def _is_on_demand_supported_search_filter(token: QueryToken) -> bool:
 
 
 def _is_excluding_transactions(token: SearchFilter) -> bool:
-    return (
+    is_not_transaction = (
         token.key.name == "event.type"
         and token.operator == "!="
         and token.value.raw_value == "transaction"
     )
+    is_error = (
+        token.key.name == "event.type"
+        and token.operator == "="
+        and token.value.raw_value == "error"
```

Comment:
I believe we also have the case for `default` (I don't know why we have it)
<img width="103" alt="image" src="https://github.com/getsentry/sentry/assets/44410/120cb245-0665-4c19-85b4-950677bd9661">

Reply:
Ok, I am now handling it the same way as `error` since for alerts we actually only look at `event.type:transaction`